File: web_backend/web_backend/Models/Badge.cs
using System.ComponentModel.DataAnnotations;

namespace web_backend.Models
{

    public class Badge
    {
        [Key]
        public int BadgeId { get; set; }

        [Required]
        [MaxLength(100)]
        public string BadgeName { get; set; }

        [Required]
        [MaxLength(255)]
        public string Description { get; set; }
        public string? ImageUrl { get; set; }

        public int? RequiredPoints { get; set; }
        public int? RequiredRecordCount { get; set; }

        [MaxLength(80)]
        public string? IconName { get; set; }

        public bool IsActive { get; set; } = true;
    }
}

File: mobil/EcoLink/app/src/main/java/com/ecolink/app/network/models/Badge.java
package com.ecolink.app.network.models;

public class Badge {
    private int badgeId;
    private String badgeName;
    private String description;
    private Integer requiredPoints;
    private Integer requiredRecordCount;
    private String iconName;
    private boolean isActive;

    public int getBadgeId() { return badgeId; }
    public String getBadgeName() { return badgeName; }
    public String getDescription() { return description; }
    public String getIconName() { return iconName; }
}
